client/dashboard/sites/emails/index.tsx

```diff
@@ -1,19 +1,205 @@
+		elements: [
+			{ value: 'mailbox', label: __( 'Mailbox' ) },
+			{ value: 'forwarding', label: __( 'Forwarder' ) },
+		],
+	},
+	{
+		id: 'status',
+		label: __( 'Status' ),
+		render: ( { item }: { item: Email } ) => {
+			if ( item.status === 'active' ) {
+				return __( 'Active' );
+			}
+			if ( item.status === 'pending' ) {
+				return __( 'Pending verification' );
+			}
+			if ( item.status === 'suspended' ) {
+				return __( 'Expired' );
+			}
+			return item.status;
+		},
+		getValue: ( { item }: { item: Email } ) => item.status,
+		// map to display values for filtering UI
+		elements: [
+			{ value: 'active', label: __( 'Active' ) },
+			{ value: 'pending', label: __( 'Pending verification' ) },
+			{ value: 'suspended', label: __( 'Expired' ) },
+		],
+	},
+];
 
 function SiteEmails() {
+	const navigate = useNavigate();
 	const { siteSlug } = siteRoute.useParams();
-	const { data: site } = useQuery( siteBySlugQuery( siteSlug ) );
+	const { data: site, isLoading: isSiteLoading } = useQuery( siteBySlugQuery( siteSlug ) );
+
+	// Wait for site to load to know ID
+	const { data: domains, isLoading: isDomainsLoading } = useQuery(
+		! isSiteLoading && site && site.ID
+			? siteDomainsQuery( site.ID )
+			: { queryKey: [ 'noop' ], queryFn: async () => [] }
+	);
+
+	const { data: allEmails, isLoading: isEmailsLoading } = useQuery( emailsQuery() );
+
+	// Filter emails to those belonging to this site by either siteId or matching one of the site's domains
+	const siteDomainNames = new Set( ( domains ?? [] ).map( ( d: SiteDomain ) => d.domain ) );
+	const emails = ( allEmails ?? [] ).filter( ( e ) => {
+		const siteIdMatch = e.siteId && Number( e.siteId ) === site?.ID;
+		const domainMatch = e.domainName && siteDomainNames.has( e.domainName );
+		return siteIdMatch || domainMatch;
+	} );
+
+	const [ selection, setSelection ] = useState< Email[] >( [] );
+	const [ view, setView ] = useState< View >( {
+		type: 'table',
+		page: 1,
+		perPage: 10,
+		sort: { field: 'emailAddress', direction: 'asc' },
+		fields: [ 'type', 'status' ],
+		titleField: 'emailAddress',
+	} );
+
+	const actions = useMemo(
+		() => [
+			{
+				id: 'view-mailbox',
+				label: __( 'View mailbox' ),
+				icon: 'admin-site',
+				callback: ( items: Email[] ) => {
+					window.open( `https://mail.${ items[ 0 ].domainName }`, '_blank' );
+				},
+				isEligible: ( item: Email ) => item.type === 'mailbox',
+			},
+			{
+				id: 'manage-subscription',
+				label: __( 'Manage subscription' ),
+				callback: () => {
+					navigate( { to: purchasesRoute.to } );
+				},
+			},
+			{
+				id: 'resend-verification',
+				label: __( 'Resend verification' ),
+				callback: () => {
+					// TODO: Wire mutation when available
+					// resendForwarderVerificationMutation.mutate({ emailId })
+				},
+				isEligible: ( item: Email ) => item.type === 'forwarding' && item.status === 'pending',
+			},
+			{
+				id: 'delete',
+				label: __( 'Delete' ),
```

Comment:
According to the Figma design, this should be either "Delete mailbox" or "Delete forwarder"